csrc/multidevice/utils.h

```diff
@@ -21,12 +22,43 @@ NVF_API bool distributedEnabled();
 // Checks that the other non-reduction axis are not parallelized on Didx
 NVF_API bool isSharded(TensorView*);
 
-// Returns the subset of tvs which elements have the same multi-device sharding
-// as ref
+// Returns the subset of tvs which elements have the different multi-device
+// sharding as ref
 template <typename TvIterator>
```

Comment:
I understand this is to have one implementation that supports both vector and FilteredView. However, it comes with the downside of increased compile time whenever this implementation is changed and multidevice/utils.h is a quite common header. You may want to consider explicit instantiation to keep the implementation in the `.cpp` file. We can also add an auto conversion from vector<T> to FilteredView<T, T::iterator>, which is likely more work.